docs/pages/reference/yaml-reference.md

```diff
@@ -148,7 +148,7 @@ This documentation effectively reflects the Java object tree under [RawServiceSp
 
     * `essential`
 
-      Marks this task as either "Essential", where relaunching this task results in relaunching all `RUNNING` tasks in the pod instance, or "Non-essential", where relaunching it does not affect other `RUNNING` tasks in the pod instance. By default this value is `true`, such that if this task needs to be relaunched, then all other tasks sharing the same pod instance are relaunched alongside it. This is only a factor in cases where a given pod has multiple `RUNNING` tasks defined.
+      Marks this task as either "Essential", where task failure results in relaunching all tasks in the pod as a unit, or "Non-essential", where task failure results in only relaunching this task and leaving other tasks in the pod unaffected. By default this value is `true`, such that the failure of the task will result in relaunching all tasks in the pod. This is only a factor in cases where a given pod has multiple `RUNNING` tasks defined.
```

Comment:
@sascala Do you want to look at this?

* `... multiple tasks with a goal state of RUNNING`
* `... only relaunching the failed task ...`